sql/analyzer/aliases.go

```diff
@@ -66,7 +66,7 @@ func getTableAliases(n sql.Node, scope *Scope) (TableAliases, error) {
 
 		if at, ok := node.(*plan.TableAlias); ok {
 			switch t := at.Child.(type) {
-			case *plan.ResolvedTable, *plan.SubqueryAlias, *plan.ValueDerivedTable:
+			case *plan.ResolvedTable, *plan.SubqueryAlias, *plan.ValueDerivedTable, *plan.TransformedNamedNode:
```

Comment:
I'm surprised that this isn't necessary in other places as well. Is that because most analyzer rules have already run before these things exist?

High time we derive a TableFactor interface for this kind of node, but that can be another day

Reply:
I think it's a combination of that and the fact that some relevant entry points are already using this interface. We do optimize indexes quite late though...